java/src/main/java/com/google/appengine/tools/mapreduce/impl/handlers/StatusHandler.java

```diff
@@ -235,6 +236,7 @@ JSONObject handleGetJobDetail(PipelineRunner pipelineRunner, ShardedJobRunId job
         shardArray.put(shardObject);
         i++;
```

Comment:
Mostly you can get rid of this, right? As now you are using a list, you can do `tasks.size()` instead of incrementing the counter each iteration